scripts/setup.cls

```diff
@@ -1,17 +1,46 @@
 public static void qaSetup() {
-  enableCourseConnections();
+    configQACustomSettings();
 }
 
-public static void enableCourseConnections() {
+//Wrapper class to config QA custom settings
+private static void configQACustomSettings() {
+    %%%NAMESPACE%%%Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
+    enableCourseConnections(orgSettings);
+    enableSpecifyRoleForCreatedAffiliations(orgSettings);
+    enableCopyAffiliationStartDateFromProgramEnrollment(orgSettings);
+    enableCopyAffiliationEndDateFromProgramEnrollment(orgSettings);  
+    upsert orgSettings;
+}
 
+//Enable course connections
+private static void enableCourseConnections(%%%NAMESPACE%%%Hierarchy_Settings__c settings) {
     Schema.DescribeSObjectResult r = Schema.SObjectType.%%%NAMESPACE%%%Course_Enrollment__c;
-  Map<String,Schema.RecordTypeInfo> rtInfosByName = r.getRecordTypeInfosByName();
+    Map<String,Schema.RecordTypeInfo> rtInfosByName = r.getRecordTypeInfosByName();
+    settings.put('%%%NAMESPACE%%%Enable_Course_Connections__c', true);
+    settings.put('%%%NAMESPACE%%%Student_RecType__c', rtInfosByName.get('Student').getRecordTypeId());
+    settings.put('%%%NAMESPACE%%%Faculty_RecType__c', rtInfosByName.get('Faculty').getRecordTypeId());
+}
 
-  %%%NAMESPACE%%%Hierarchy_Settings__c settings = %%%NAMESPACE%%%Hierarchy_Settings__c.getOrgDefaults();
+//Enable Specify Role For Created Affiliations
+private static void enableSpecifyRoleForCreatedAffiliations(%%%NAMESPACE%%%Hierarchy_Settings__c settings) {   
+    if (settings.%%%NAMESPACE%%%Affl_ProgEnroll_Set_Role__c == null 
+        || settings.%%%NAMESPACE%%%Enable_New_Preferred_Phone_Sync__c != true) {
```

Comment:
This is a different field from the null check in this conditional. Is that intentional?